patches/tModLoader/Terraria/Main.cs.patch

```diff
@@ -1115,6 +1115,19 @@
  		ClientInitialize();
  		base.Initialize();
  	}
+@@ -5242,7 +_,11 @@
+ 		Chest.SetupTravelShop();
+ 		for (int m = 1; m < 100; m++) {
+ 			shop[m] = new Chest();
+-			shop[m].SetupShop(m);
++
++			if (m >= NPCLoader.shopToNPC.Length)
++				continue;
++
++			shop[m].SetupShop(NPCShopDatabase.GetNpcShopByIndex(m), m);
```

Comment:
Is this necessary? Can we just leave `shop` full of blank `new Chest()` values until the player opens one of them, since we rebuild every time anyway?